FreeTubeApp/FreeTube
# Add RYD and RYD Proxy support

# Add RYD and RYD Proxy support

## Pull Request Type

## Related issue
closes https://github.com/FreeTubeApp/FreeTube/issues/1927
Credit to @petaded for https://github.com/FreeTubeApp/FreeTube/pull/1982

## Description
This PR adds support for the ReturnYouTubeDislikes api as well a RYD proxies 
https://github.com/TeamPiped/RYD-Proxy

## Testing 
RYD Disabled - (Test with local and invidious)
1) have ryd disabled 
2) test a video (ex: https://www.youtube.com/watch?v=YbJOTdZBX1g )
2) dislikes aren't fetched and aren't shown

RYD Enabled + using RYD-proxy - (Test with local and invidious)
1) have ryd enabled
2) use ryd-proxy (ex: kavin's instance [this is the default instance])
3) test a video (ex: https://www.youtube.com/watch?v=YbJOTdZBX1g )
4) see dislikes + like/dislike bar

RYD Enabled + using official API - (Test with local or invidious)
1) have ryd enabled
2) use official api 
3) test a video (ex: https://www.youtube.com/watch?v=YbJOTdZBX1g )
4) see dislikes + like/dislike bar

## Desktop
- **OS:** Windows
- **OS Version:** 11
- **FreeTube version:** 0.18.0
